packages/react-storage/src/components/StorageBrowser/context/actions/downloadAction.ts

```diff
@@ -15,12 +15,18 @@ export function downloadAction(
 
   const bucket = bucketName && region ? { bucketName, region } : undefined;
 
-  downloadData({
-    path,
-    options: {
-      bucket,
-      locationCredentialsProvider: credentialsProvider,
-    },
-  });
-  return Promise.resolve({ key: path });
+  try {
+    const signedUrl = await getUrl({
+      path,
+      options: {
+        bucket,
+        locationCredentialsProvider: credentialsProvider,
+        validateObjectExistence: true,
```

Comment:
Added this `validateObjectExistence`, which throws an error if the item doesn't exist anymore so that the download link doesn't try to download something that doesn't exist.

At the moment if you try to download something that doesn't exist, there is no UI update to let the user know that the file they were trying to download doesn't exist. Can only see the error the dev tools console or network tab